u-root.go

```diff
@@ -136,11 +147,74 @@ func generateLabel() string {
 	return fmt.Sprintf("%s-%s-%s-%s", *build, env.GOOS, env.GOARCH, strings.Join(baseCmds, "_"))
 }
 
+func gobusyboxMain() error {
+	bopts := &golang.BuildOpts{}
+	// hmm.
+	//	bopts.RegisterFlags(flag.CommandLine)
+	flag.Parse()
+
+	// Why doesn't the log package export this as a default?
+	l := log.New(os.Stdout, "", log.LstdFlags)
+
+	o, err := filepath.Abs(*outputPath)
+	if err != nil {
+		return err
+	}
+
+	env := golang.Default()
+	if env.CgoEnabled {
+		l.Printf("Disabling CGO for u-root...")
+		env.CgoEnabled = false
+	}
+	l.Printf("Build environment: %s", env)
+
+	tmpDir := *genDir
+	remove := false
+	if tmpDir == "" {
+		tdir, err := ioutil.TempDir("", "bb-")
+		if err != nil {
+			l.Fatalf("Could not create busybox source directory: %v", err)
+		}
+		tmpDir = tdir
+		remove = true
+	}
+
+	opts := &bb.Opts{
+		Env:          env,
+		GenSrcDir:    tmpDir,
+		CommandPaths: flag.Args(),
+		BinaryPath:   o,
+		GoBuildOpts:  bopts,
+	}
+	if err := bb.BuildBusybox(opts); err != nil {
+		l.Print(err)
+		var errGopath *bb.ErrGopathBuild
+		var errGomod *bb.ErrModuleBuild
+		if errors.As(err, &errGopath) {
+			return fmt.Errorf("Preserving bb generated source directory at %s due to error. To reproduce build, `cd %s` and `GO111MODULE=off GOPATH=%s go build`.", tmpDir, errGopath.CmdDir, errGopath.GOPATH)
+		} else if errors.As(err, &errGomod) {
+			return fmt.Errorf("Preserving bb generated source directory at %s due to error. To debug build, `cd %s` and use `go build` to build, or `go mod [why|tidy|graph]` to debug dependencies, or `go list -m all` to list all dependency versions.", tmpDir, errGomod.CmdDir)
+		} else {
+			return fmt.Errorf("Preserving bb generated source directory at %s due to error.", tmpDir)
+		}
+	}
+	// Only remove temp dir if there was no error.
+	if remove {
+		os.RemoveAll(tmpDir)
+	}
+}
+
 func main() {
 	flag.Parse()
 
 	start := time.Now()
 
+	if *usegobusybox {
```

Comment:
Can you enable this option in the uroot_test.go file from now on?

Reply:
one war at a time :-)